PYTHON_STYLE_GUIDE.md

```diff
@@ -0,0 +1,79 @@
+
+* [Google's docs on Google-style docstrings](https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments)
+* [napoleon's docs include an overview of Google-style docstrings](http://sphinxcontrib-napoleon.readthedocs.org/en/latest/index.html)
+* [Example Google-style docstrings](http://sphinxcontrib-napoleon.readthedocs.org/en/latest/example_google.html) (from napoleon's docs)
+
+### Maximum Line Length
+
+PEP8 has some [maximum line length guidelines](https://www.python.org/dev/peps/pep-0008/#id17), starting with "Limit all lines to a maximum of 79 characters" but "for flowing long blocks of text with fewer structural restrictions (docstrings or comments), the line length should be limited to 72 characters."
+
+We discussed this at length, and it seems that the consensus is: try to keep line lengths less than 79/72 characters, unless you have a special situation where longer lines would improve readability. (The basic reason is that 79/72 works for everyone, and BigchainDB is an open source project.)
+
+### Single or Double Quotes?
+
+Python lets you use single or double quotes. PEP8 says you can use either, as long as you're consistent. We try to stick to using single quotes, except in cases where using double quotes is more readable. For example:
+```python
+print('This doesn\'t look so nice.')
+print("Doesn't this look nicer?")
+```
+
+### Breaking Strings Across Multiple Lines
+
+Should we use parentheses or slashes (`\`) to break strings across multiple lines, i.e.
+```python
+my_string = ('This is a very long string, so long that it will not fit into just one line '
+             'so it must be split across multiple lines.')
+# or
+my_string = 'This is a very long string, so long that it will not fit into just one line ' \
+            'so it must be split across multiple lines.'
+```
+
+It seems the preference is for slashes, but using parentheses is okay too. (There are good arguments either way. Arguing about it seems like a waste of time.)
+
+### Using the % operator or `format()` to Format Strings
+
+Given the choice:
+```python
+x = 'name: %s; score: %d' % (name, n)
+# or
+x = 'name: {}; score: {}'.format(name, n)
+```
+
+we use the `format()` version. It's newer and doesn't require you to remember a bunch of arcane markup.
+
+
+## Writing (Python) Tests
+
+We write tests for our Python code using the [pytest](http://pytest.org/latest/) framework.
+
+All tests go in the `bigchaindb/tests` directory or one of its subdirectories. You can use the tests already in there as templates or examples.
+
+To run all the tests, first make sure you have RethinkDB running:
+```text
+$ rethinkdb
+```
+
+then in another terminal, do:
+```text
+$ py.test -v
+```
+
+If that doesn't work (e.g. maybe you are running in a conda virtual environment), try:
+```text
+$ python -m pytest -v
+```
+
```

Comment:
perhaps we can also mention the use of `docker-compose` to run the tests ... i.e.:

Build the images (~ once):

```bash
$ docker-compose build
```

Run the tests (`rethinkdb` is taken care of by `docker-compose` via the `docker-compose.yml` script)
```
$ docker-compose run --rm bigchaindb py.test -v
```